From 3cce798cc4849594390629180bf3ba56c6c0c3e2 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Fri, 29 Nov 2019 14:20:20 -0800
Subject: [PATCH] rclcpp_components-extras.cmake.in: use path relative to
 rclcpp_components_DIR in @PROJECT_NAME@_NODE_TEMPLATE

* without this the resulting path is absolute path like this:
  /usr/share/rclcpp_components/node_main.cpp.in
  and building something which calls configure_file on this fails with:
  CMake Error: File /usr/share/rclcpp_components/node_main.cpp.in does not exist.
  when cross-compiling with OE, because the rclcpp_components are
  installed in RSS prefix (e.g.
  /jenkins/mjansa/build-ros-thud-mcf-herb/artifacts/BUILD-ros2-eloquent-thud.mcf/work/i586-oe-linux/action-tutorials-cpp/0.8.4-1-r0/recipe-sysroot/usr/share/rclcpp_components/node_main.cpp.in

* the simplest fix is to use rclcpp_components_DIR which is set
  correctly from CMake

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 rclcpp_components-extras.cmake.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rclcpp_components-extras.cmake.in b/rclcpp_components-extras.cmake.in
index f1956c1..6815096 100644
--- a/rclcpp_components-extras.cmake.in
+++ b/rclcpp_components-extras.cmake.in
@@ -25,7 +25,7 @@ macro(_rclcpp_components_register_package_hook)
   endif()
 endmacro()
 
-set(@PROJECT_NAME@_NODE_TEMPLATE "@CMAKE_INSTALL_PREFIX@/@node_main_template_install_dir@/node_main.cpp.in")
+set(@PROJECT_NAME@_NODE_TEMPLATE "${rclcpp_components_DIR}/../node_main.cpp.in")
 
 include("${rclcpp_components_DIR}/rclcpp_components_register_nodes.cmake")
 include("${rclcpp_components_DIR}/rclcpp_components_register_node.cmake")
